### ISSUE TITLE
What does Internal error: TLS::Channel No read cipherstate for epoch 1 means?

### ISSUE THREAD

**Issue by jeffRTC** [randombit/botan#2675]
What does this error means and why it happens?

Parent issue:  https://github.com/randombit/botan/issues/2666

This error happened right after I made Botan::TLS::Server global (It was created for each connection before)

**Comment by randombit** [comment]
Basically that we've received a ciphertext that is encrypted but the handshake hasn't completed yet.

> This error happened right after I made Botan::TLS::Server global (It was created for each connection before)

This would certainly do it - the `Server` is the *per connection* state.

**Comment by jeffRTC** [comment]
@randombit  If Server is for per connection then it seems there is certainty bug in your library. I can't test your Asio thing because of LetsEncrypt private key (It doesn't have a password so I can't convert it into Botan format).

I've searched all over Github to find a example that uses Raw Sockets (Not Asio) and Botan,

https://github.com/zelinliu2/template_login/blob/5c8e46596bc70d64b02fbdfd79a9899cc21564c0/tests/tlstest/botantest/tlstest_server.cpp

I found one and I don't find any difference with that one versus mine.
